snap/validate.go

```diff
@@ -562,20 +564,18 @@ func validateAppOrderCycles(apps map[string]*AppInfo) error {
 func validateAppOrderNames(app *AppInfo, dependencies []string) error {
 	// we must be a service to request ordering
 	if len(dependencies) > 0 && !app.IsService() {
-		return fmt.Errorf("cannot define before/after in application %q as it's not a service", app.Name)
+		return errors.New("must be a service to define before/after ordering")
 	}
 
 	for _, dep := range dependencies {
 		// dependency is not defined
 		other, ok := app.Snap.Apps[dep]
 		if !ok {
-			return fmt.Errorf("application %q refers to missing application %q in before/after",
-				app.Name, dep)
+			return fmt.Errorf("refers to missing application %q in before/after", dep)
```

Comment:
+1 for changing the original message, but the new version feels like it's missing a subject. How about: `reference to a missing application ....`? Perhaps @degville  can advise again.

Reply:
+1 on @stolowski's suggestion, or how about `before/after references a missing application`?